doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,255 @@
+given to any class; ``auto`` - where weight is given to a sample based
+on its class distribution in the dataset; and a dictionary of the form
+``{class_label : class_weight}``.
+
+Figure 2 shows the result of adding weight to the minority class, where the 
+decision boundary surrounding that class becomes larger in radius as more weight is given.
+
+.. figure:: ../auto_examples/neural_networks/images/plot_weighted_elm_001.png
+   :target: ../auto_examples/neural_networks/plot_weighted_elm.html
+   :align: center
+   :scale: 100%
+
+   Figure 2 : Effect of sample weighting on an imbalanced dataset.
+
+.. topic:: Examples:
+
+   * :ref:`example_neural_networks_plot_weighted_elm.py`
+ 
+
+.. _elm_regression:
+
+Regression
+==========
+
+:class:`ELMRegressor` can solve regression problems consisting of one or more
+output values assigned to each sample. It differs from :class:`ELMClassifier`, 
+in that the final output is a continuous value.
+
+The fit method in :class:`ELMRegressor` accepts arguments X and y
+where y is expected to be a matrix of floating point values::
+
+    >>> from sklearn.neural_network import ELMRegressor
+    >>> X = [[0, 0], [2, 2]]
+    >>> y = [0.5, 2.5]
+    >>> clf = ELMRegressor(random_state=1)
+    >>> clf.fit(X, y)
+    ELMRegressor(C=1, activation='relu', batch_size=None, n_hidden=100,
+           random_state=1, verbose=False, warm_start=False, weight_scale=1.0)
+
+    >>> clf.predict([[1, 1]])
+    array([ 1.30454982])
+
+
+.. _elm_tips:
+
+Tips on Practical Use
+=====================
+
+  * **Setting n_hidden**: increasing ``n_hidden`` - the number of nodes in 
+    the hidden layer, allows the ELM to capture more complex representations of
+    the input features, usually leading to better generalization. However, higher
+    ``n_hidden`` value (1) would require ``C`` and ``weight_scale`` to be adjusted
+    to avoid potential overfitting, and (2) would increase memory usage, 
+    training time, and prediction time.
+
+  * **Setting C**: if the dataset has few samples or if the samples are noisy, 
+    decreasing ``C`` will increase regularization which in turn reduces 
+    overfitting and might improve generalization. Increasing ``C`` will 
+    therefore decrease regularization.
+
+  * **Setting weight_scale**:  this controls regularization of the weights 
+    the same way as ``C`` does; but instead of regularizing the hidden-to-output 
+    layer weights, it regularizes the input-to-hidden layer weights.
+
+  * **Setting class_weight**: if the dataset contains an imbalanced ratio 
+    of class fequencies, it is desirable to set `class_weight='auto'`, as
+    it assigns more weight to the class having less number of samples.
+
+  * **Setting batch_size**: training on large datasets or having a large 
+    ``n_hidden`` value could cause issues for limited memory. In this case, it is 
+    recommended to set ``batch_size=1000`` in order to process the data in chunks. 
```

Comment:
Maybe you should say here that by default, all samples are processed as a whole. And recommend setting "an upper limit such as ``batch_size=1000``".